src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal partial class DocumentOutlineViewModel : INotifyPropertyChanged, IDisposable
```

Comment:
this does not feel like a model to me.

Reply:
ViewModels are expected to always implement `INotifyPropertyChanged` in the WPF world and this is a pattern that we follow throughout roslyn.

- [RenameDashboardViewModel](https://github.com/dotnet/roslyn/blob/54a50247a9f875916947815176dacadf2d7444ee/src/EditorFeatures/Core.Wpf/InlineRename/UI/Dashboard/RenameDashboardViewModel.cs#L18)
- [AbstractOptionPreviewViewModel](https://github.com/dotnet/roslyn/blob/54a50247a9f875916947815176dacadf2d7444ee/src/VisualStudio/Core/Impl/Options/AbstractOptionPreviewViewModel.cs#L35)
- [ViewModelBase](https://github.com/dotnet/roslyn/blob/54a50247a9f875916947815176dacadf2d7444ee/src/VisualStudio/Core/Def/Utilities/ViewModelBase.cs#L15)